server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -160,7 +160,8 @@ public static State fromString(String state) {
         "index.number_of_routing_shards",
         INDEX_NUMBER_OF_SHARDS_SETTING,
         1,
-        new Setting.Validator<>() {
+        //noinspection Convert2Diamond since some IntelliJs mysteriously report an error if the <Integer> is replaced with <> here:
```

Comment:
This suppression doesn't work on my IDEA for some reason, what does work however is 

```java
    @SuppressWarnings("Convert2Diamond")
    public static final Setting<Integer> INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING = Setting.intSetting(
```

maybe that work for yours as well? :D

Reply:
Weirder and weirder. Yeah that works for me too.